astropy/nddata/utils.py

```diff
@@ -456,3 +469,322 @@ def block_replicate(data, block_size, conserve_sum=True):
         data = data / float(np.prod(block_size))
 
     return data
+
+
+class Cutout2D(object):
+    def __init__(self, data, position, shape, wcs=None, mode='trim',
+                 fill_value=np.nan, copy=False):
+        """
+        Create a cutout object from a 2D array.
+
+        The returned object will contain a 2D cutout array.  If
+        ``copy=False`` (default), the cutout array is a view into the
+        original ``data`` array, otherwise the cutout array will contain
+        a copy of the original data.
+
+        If a `~astropy.wcs.WCS` object is input, then the returned
+        object will also contain a copy of the original WCS, but updated
+        for the cutout array.
+
+        For example usage, see :ref:`cutout_images`.
+
+        Parameters
+        ----------
+        data : `~numpy.ndarray`
+            The 2D data array from which to extract the cutout array.
+
+        position : tuple or `~astropy.coordinates.SkyCoord`
+            The position of the cutout array's center with respect to
+            the ``data`` array.  The position can be specified either as
+            a ``(x, y)`` tuple of pixel coordinates or a
+            `~astropy.coordinates.SkyCoord`, in which case ``wcs`` is a
+            required input.
+
+        shape : tuple
+            The shape (``(ny, nx)``) of the cutout array in pixel
```

Comment:
It's certainly possible.  We do this exact thing for the `photutils` sky apertures.  To prevent duplication, the functions in `photutils.utils.wcs_helpers` would first need to be moved to `astropy`.  I prefer to do this in a separate PR, so I'm going to pass on this for now.